### ISSUE TITLE
Accessibility: missing labels

### ISSUE THREAD

**Issue by nullatonce** [laurent22/joplin#5603]
## Environment

Joplin version:  2.4.12 
Platform:  Windows 10
OS specifics: 

## Steps to reproduce

1.start screen reader
2. start joplin
3.  navigate with keyboard

## Describe what you expected to happen
Elements to have names so screen readers would work

## DESCRIPTION:

To keep it simpler all of my navigation writing this issue was with arrow down.
WSR - built in Windows Screen Reader (ctrl+win+enter)
I noticed that WSR and NVDA ( Non Visual Desktop Enviroment) on some elements act differently, if no screen reader specified assume that behavior is the same.

1.First element that is pronounced is "Notebooks" moving to next element screen reader reads: "button"

2.Moving to next element after "button":
WSR: link;
NVDA: clickable;
(This is for each list item of Notebooks)
3. After 2there is a link of "All notes" (works, mentioned as reference).
4. Element aAfter 3:
NVDA:clickable
WSR: link

5. Button after "Search" has no label *so assuming it's to search).

6. After  "New To-Do" and "New Note" checkbox has no label

**Comment by mak2002** [comment]
Hey there @nullatonce , I am new here and want to start contributing. Is this issue good for newbie if yes I would like to work on this

**Comment by vgjh2005** [comment]
Hi @mak2002
Still here. Please fix it!

**Comment by personalizedrefrigerator** [comment]
2024 status update:

<blockquote>

1. First element that is pronounced is "Notebooks" moving to next element screen reader reads: "button"

</blockquote>

The element just after "Notebooks" is now read as "New, pushbutton" ([changed here](https://github.com/laurent22/joplin/commit/97b5276f81ae2aaee321ef4d0554271935ca53fe#diff-56b419dd028a92d8aa53156607030081749ca3c972c2c435241ae28a3e431168R27)). (Orca screen reader on Ubuntu 24.04).

<blockquote>
2. Moving to next element after "button":
WSR: link;
NVDA: clickable;
(This is for each list item of Notebooks)
</blockquote>

Moving to the next element after "New, pushbutton":
- "Symbol e908 All notes, link"

For the next item (for me, a notebook called "scheduling" with an alarmclock icon):
- "Expand Scheduling link. Alarmclock scheduling link. Comment link 138."

Other notebooks are read similarly. Notebooks with no subnotebooks don't have the "Expand notebook name here link".

<blockquote>
4. After 2 there is a link of "All notes" (works, mentioned as reference).
5. Element aAfter 3:
NVDA:clickable
WSR: link

   Button after "Search" has no label *so assuming it's to search).

</blockquote>

The button is now read as "Search pushbutton" when not searching. Focusing it and pressing <kbd>space</kbd> focuses the search input. While searching, the button reads "Clear search pushbutton".

<blockquote>

   After "New To-Do" and "New Note" checkbox has no label

</blockquote>

After creating a task with "new to-do", then focusing it in the note list (<kbd>ctrl</kbd>-<kbd>shift</kbd>-<kbd>l</kbd>) the checkbox for the to-do is difficult to focus and still has no label.

**Comment by laurent22** [comment]
Closed in favour of https://github.com/laurent22/joplin/issues/10795
